app/stores/CollectionsStore.js

```diff
@@ -135,6 +159,12 @@ export default class CollectionsStore extends BaseStore<Collection> {
     if (path) return path.title;
   }
 
+  getByUrl(url: string): ?Collection {
+    return find(Array.from(this.data.values()), (col: Collection) =>
```

Comment:
```suggestion
    return find(this.orderedData, (col: Collection) =>
```

`orderedData` is already a pre-computed array we can use here